A-4E-C/Cockpit/Scripts/Systems/gear.lua

```diff
@@ -89,7 +89,92 @@ dev:listen_command(device_commands.emer_gear_release)
 dev:listen_command(NWS_Engage)
 dev:listen_command(NWS_Disengage)
 
+-- Skiding detection window
+local no_anim_L_array = {}
+local no_anim_R_array = {}
+local no_anim_N_array = {}
+
+for i=1, 10 do
+    no_anim_L_array[i] = -1
+    no_anim_R_array[i] = -1
+    no_anim_N_array[i] = -1
+end
+local skid_L_param = get_param_handle("SKID_L_DETECTOR")
+local skid_R_param = get_param_handle("SKID_R_DETECTOR")
+local skid_N_param = get_param_handle("SKID_N_DETECTOR")
+
+function get_base_sensor_data()
```

Comment:
Why is this here? Maybe you are looking for a similar function which can be found in EFM_data_bus

Reply:
Oh, it is here because I copy-paste the ```on_carrier``` function from ```Cockpit\Scripts\Systems\carrier.lua``` (with some modifications), and this function was defined inside it (```carrier.lua```). After review, only using the default ```get_base_data``` from EFM_data_bus should be fine, let me modify it.

Reply:
Done, let me know if it is now ok for you (tested in quick missions on land and aircraft carriers)

Reply:
You can just do something like: 

`local sensor_data = get_efm_sensor_data_overrides()`

at the top. sensor_data will just be a table of functions so you only need to call it once to get the table.

Then wherever you want the position
`local px,py,pz = sensor_data.getSelfCoordinates()`

This would simplifiy this.

Reply:
You actually already have and use a sensor_data at the top see line 27

Reply:
Got it, have made the change to use the already declared sensor_data